### PR TITLE
feat: migrated and cleaned up logical data model documentation

### PR DESCRIPTION
**Pull request by meekg33k**
- Migrated the `Logical Data Model` documentation.
  - Moved content from existing documentation to Vuepress documentation.
  - Created sidebar config for `Logical Data Model` documentation.

- Cleaned up and restructured the `Logical Data Model` documentation.
  - Fixed typos and made grammatical errors.
  - Replaced numbered headers and sub-headers with named headers. e.g. `## 4.1 ABC` to `## ABC` to allow for ease of modification.
  - Restructured document structure and moved `List of Tables` to the end of the documentation.
  - Added unique section header links to avoid link duplication.
  - Updated table links so that they point to the tables themselves not the section in which they are contained.

Screenshot of `Logical Data Model` documentation: 

updated-logical-data-model

### CODE REVIEW

**Comment by ehenrka** [comment]
Some general comments/questions:

- The screenshot includes the text "Section 3 lists ..." and "Section 4 describes ...", but these should have been removed as far as I can understand? I'm fine with the change to remove section numbers, as long as we add a link to the actual section instead anywhere it is referenced.
- In the sidebar to the left, which level of sections can you see? I think I can see \# up to \#\#\# from the screenshot. Can you see \#\#\#\# as well? Would be great if it was collabsible, so that you could extend it to see sections on \#\#\#\# as well.
- I'm fine with moving _List of tables_ and similar to the end.
- I would like to see a lot more links, for example if I click on PartyIdType then I would like to get to the definition of that (the missing links in this document is not related to this change). I as a reader should not have to manually search for the definition of PartyIdType in the document, especially as the word PartyIdType is used a lot which means that it is very hard to find.

Then I have some more comments regarding the document, which should not be related to the changes here.
- There are a lot of **Error! Reference source not found.** in the document (comes from Word when updating references).
- There are some instances of a table within another table, see for example Table 10 (this error is also from the Word document when the references are incorrectly updated).

**Comment by meekg33k** [comment]
> Some general comments/questions:
> 
> * The screenshot includes the text "Section 3 lists ..." and "Section 4 describes ...", but these should have been removed as far as I can understand? I'm fine with the change to remove section numbers, as long as we add a link to the actual section instead anywhere it is referenced.
> * In the sidebar to the left, which level of sections can you see? I think I can see # up to ### from the screenshot. Can you see #### as well? Would be great if it was collabsible, so that you could extend it to see sections on #### as well.
> * I'm fine with moving _List of tables_ and similar to the end.
> * I would like to see a lot more links, for example if I click on PartyIdType then I would like to get to the definition of that (the missing links in this document is not related to this change). I as a reader should not have to manually search for the definition of PartyIdType in the document, especially as the word PartyIdType is used a lot which means that it is very hard to find.
> 
> Then I have some more comments regarding the document, which should not be related to the changes here.
> 
> * There are a lot of **Error! Reference source not found.** in the document (comes from Word when updating references).
> * There are some instances of a table within another table, see for example Table 10 (this error is also from the Word document when the references are incorrectly updated).

@ehenrka, as per your comments above:

- I've updated the text to reflect, see screenshot below. See the updated screenshot in the MR description.

- For displaying headers, Vuepress has a config option called `sidebarDepth` that allows you specify the links for headers on a page. The max value of 2, which is currently what we are using only extracts `h2` and `h3` headers. See more details [here](https://vuepress.vuejs.org/theme/default-theme-config.html#nested-header-links). 

- For linking the different resources in the document, I totally agree and I had already started some work on that. See screenshots below. 

**Old (Gitbooks)**
gitbooks-api-supporting-model

**Current (Vuepress)**
vuepress-api-supporting-model

However, I think that will be beyond the scope of this MR so I have created a separate issue [#2545](https://github.com/mojaloop/project/issues/2545) to track that work.

With regards to the other points you made:

- I've updated Tables 10 and 11 so they no longer show the incorrectly nested tables (see screenshot below). I had to do this for about 6 of these tables.

fix-for-table-10

- Yes there is a lot of **Error! Reference source not found.** in the document and it also appears in the current documentation [here](https://docs.mojaloop.io/mojaloop-specification/fspiop-api/documents/Logical%20Data%20Model.html#431-amounttype). This was something I wanted to bring up with @lewisdaly and @elnyry-sam-k , so if you have any ideas on how to fix that.

**Comment by ehenrka** [comment]
Thanks @meekg33k, then I think we are aligned. It would be good to test using `sidebarDepth: 3`, possibly together with `collapsable: true`, to be able to quickly find sections from the sidebar, but this can probably be fine-tuned later on.

Regarding the **Error! Reference source not found**, these comes from the Word document that was used to generate the markdown file. If we take Table 39 (_Element AmountType_) as an example, the correct link/reference is actually in the **Description** column (the `AmountType`, https://github.com/mojaloop/documentation/blob/90931f7b8d76f649493c358ffe3bb20e6f558a4e/vuepress/docs/api/fspiop/logical-data-model.md#amounttype-enum), its just not correct in the **Type** column. The same actually seems to be true for all of the instances of the error, that the correct link is in the Description.

**Comment by ehenrka** [comment]
Yes @meekg33k, I'm very well aware that they are not links now :). But, they should have contained links if we go back to the original source document.

The data model document is a logical version of the data model in _API Definition_ document for the FSP Interoperability API. This document in its original form (PDF exported from Word) can be found at: https://github.com/mojaloop/mojaloop-specification/raw/master/fspiop-api/documents/v1.1-document-set/API%20Definition_v1.1.pdf. If we look at `AmountType` in Section 7.3.1, Table 45, in that document (the PDF version), then there are two links in the **Format** column, one for `Enum` and one for `String` so that is it easy to find the definition of these two elements just by clicking. It is the reference to the `Enum` that has failed to be updated in the original Word version of the _Logical Data Model_ document.

If you have a reference in Word and then update the references in Word, if the original reference can't be found then you get the "**Error! Reference source not found**".

During the conversion to Markdown format, both these references should have become links, but unfortunately that seems to have been lost in the conversion (https://docs.mojaloop.io/mojaloop-specification/fspiop-api/documents/API%20Definition%20v1.1.html).

In the _Logical Data Model_ document, the AmountType definition is in Table 39. The text in the Format column should say: "Enum of String(1..32)", and "Enum" should be a link to the definition of an Enum (https://github.com/mojaloop/documentation/blob/e7e3e3525d90f3818f362b59c0afab063bfd9240/vuepress/docs/api/fspiop/logical-data-model.md#enum) and "String" should be a link to the definition of a String (https://github.com/mojaloop/documentation/blob/e7e3e3525d90f3818f362b59c0afab063bfd9240/vuepress/docs/api/fspiop/logical-data-model.md#string).

**Comment by meekg33k** [comment]
@ehenrka, @elnyry-sam-k can I get a final review and approval of this PR?

File: vuepress/docs/api/fspiop/logical-data-model.md [hunk 1]
```diff
@@ -0,0 +1,2437 @@
...
+
+<br />
+
+#### AuthenticationType element
+
+Table 40 below contains the data model for the element `AuthenticationType`.
+
+| Name | Cardinality | Type | Description |
+| --- | --- | --- | --- |
+| **Authentication** | 1 | [Enum](#enum) of [String(1..32)](#string) | This element contains the authentication type. See [AuthenticationType](#authenticationtyype-enum) enumeration for possible enumeration values. |
```
Comment (ehenrka) [inline vuepress/docs/api/fspiop/logical-data-model.md:1216]: > | **Authentication** | 1 | [Enum](#enum) of [String(1..32)](#string) | This element contains the authentication type. See [AuthenticationType](#authenticationtyype-enum) enumeration for possible enumeration values. |

`authenticationtyype-enum` should be `authenticationtype-enum`

**Comment by ehenrka** [comment]
We should have a script to automatically check all links to be valid, to avoid problems like comment above. Otherwise I'm happy.

File: vuepress/docs/api/fspiop/logical-data-model.md (hunk 1 above)
Comment (meekg33k) [reply vuepress/docs/api/fspiop/logical-data-model.md]: Corrected the links typo.

**Comment by meekg33k** [comment]
> We should have a script to automatically check all links to be valid, to avoid problems like comment above. Otherwise I'm happy.

Agreed, it's something I'll look into.

**Pull request merged**